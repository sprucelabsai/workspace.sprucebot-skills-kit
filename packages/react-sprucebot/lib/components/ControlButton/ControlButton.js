'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Icon = require('../Icon/Icon');

var _Icon2 = _interopRequireDefault(_Icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var StyledIconLeft = (0, _styledComponents2.default)(_Icon2.default).withConfig({
	displayName: 'ControlButton__StyledIconLeft',
	componentId: 'jro3dx-0'
})(['margin-right:0.25em;']);

var StyledIconRight = (0, _styledComponents2.default)(_Icon2.default).withConfig({
	displayName: 'ControlButton__StyledIconRight',
	componentId: 'jro3dx-1'
})(['margin-left:0.25em;']);

var Button = _styledComponents2.default.button.attrs({
	className: function className(_ref) {
		var _className = _ref.className;
		return 'ControlButton control-button ' + (_className || '');
	}
}).withConfig({
	displayName: 'ControlButton__Button',
	componentId: 'jro3dx-2'
})(['align-items:center;background:none;color:#00aac7;display:inline-flex;font-weight:normal;letter-spacing:0;position:relative;margin:0;padding:0;width:auto;&:hover{background:none;}']);

var Link = _styledComponents2.default.a.attrs({
	className: function className(_ref2) {
		var _className2 = _ref2.className;
		return 'ControlButton control-button ' + (_className2 || '');
	}
}).withConfig({
	displayName: 'ControlButton__Link',
	componentId: 'jro3dx-3'
})(['align-items:center;color:#00aac7;cursor:pointer;display:inline-flex;position:relative;text-decoration:none;']);

var Wrapper = function Wrapper(_ref3) {
	var href = _ref3.href,
	    props = _objectWithoutProperties(_ref3, ['href']);

	if (href && href != '') {
		return _react2.default.createElement(Link, _extends({ href: href }, props));
	} else {
		return _react2.default.createElement(Button, props);
	}
};

var ControlButton = function ControlButton(_ref4) {
	var iconLeft = _ref4.iconLeft,
	    iconRight = _ref4.iconRight,
	    onClick = _ref4.onClick,
	    children = _ref4.children,
	    props = _objectWithoutProperties(_ref4, ['iconLeft', 'iconRight', 'onClick', 'children']);

	return _react2.default.createElement(
		Wrapper,
		_extends({
			onClick: onClick,
			iconLeft: iconLeft,
			iconRight: iconRight
		}, props, {
			tabIndex: 0
		}),
		iconLeft && _react2.default.createElement(
			StyledIconLeft,
			null,
			iconLeft
		),
		_react2.default.createElement(
			'span',
			null,
			children
		),
		iconRight && _react2.default.createElement(
			StyledIconRight,
			null,
			iconRight
		)
	);
};

ControlButton.propTypes = {
	onClick: _propTypes2.default.func,
	children: _propTypes2.default.any,
	className: _propTypes2.default.string,
	iconLeft: _propTypes2.default.string,
	iconRight: _propTypes2.default.string
};

exports.default = ControlButton;