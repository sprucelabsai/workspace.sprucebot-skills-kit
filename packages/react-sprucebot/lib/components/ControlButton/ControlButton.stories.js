'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _react3 = require('@storybook/react');

var _addonActions = require('@storybook/addon-actions');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _LinkPile = require('../LinkPile/LinkPile');

var _LinkPile2 = _interopRequireDefault(_LinkPile);

var _ControlButton = require('./ControlButton');

var _ControlButton2 = _interopRequireDefault(_ControlButton);

var _ControlButton3 = require('./ControlButton.md');

var _ControlButton4 = _interopRequireDefault(_ControlButton3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var iconOptions = {
	null: 'None',
	favorite: 'Favorite',
	edit: 'Edit',
	close: 'Close',
	keyboard_arrow_up: 'Up',
	keyboard_arrow_down: 'Down',
	chevron_left: 'Left',
	chevron_right: 'Right',
	'ðŸ¤–': 'Bot'
};

var defaultIcon = 'null';

var stories = (0, _react3.storiesOf)('ControlButton', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_ControlButton4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_LinkPile2.default,
		null,
		_react2.default.createElement(
			_ControlButton2.default,
			{
				onClick: (0, _addonActions.action)('ControlButton onClick')
			},
			'Control all the buttons!'
		),
		_react2.default.createElement(
			_ControlButton2.default,
			{
				iconLeft: (0, _react4.text)('Left Icon', 'edit'),
				onClick: (0, _addonActions.action)('ControlButton onClick')
			},
			'I have a left icon'
		),
		_react2.default.createElement(
			_ControlButton2.default,
			{
				iconLeft: (0, _react4.text)('Left Icon', 'edit'),
				iconRight: (0, _react4.text)('Right Icon', 'favorite'),
				onClick: (0, _addonActions.action)('ControlButton onClick')
			},
			'I have a right Icon'
		),
		_react2.default.createElement(
			_ControlButton2.default,
			{
				iconRight: (0, _react4.text)('Rigth Icon', 'ðŸ¤–'),
				href: (0, _react4.text)('Link href', 'https://sprucebot.com'),
				onClick: (0, _addonActions.action)('Link onClick')
			},
			'I\'m a link!'
		)
	);
})));