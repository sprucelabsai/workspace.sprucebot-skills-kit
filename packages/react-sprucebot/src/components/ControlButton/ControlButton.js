import styled from 'styled-components'
import PropTypes from 'prop-types'
import React from 'react'

import Icon from '../Icon/Icon'

const StyledIconLeft = styled(Icon)`
	margin-right: 0.25em;
`

const StyledIconRight = styled(Icon)`
	margin-left: 0.25em;
`

const Button = styled.button.attrs({
	className: ({ className }) =>
		`ControlButton control-button ${className || ''}`
})`
	align-items: center;
	background: none;
	color: #00aac7;
	display: inline-flex;
	font-weight: normal;
	letter-spacing: 0;
	position: relative;
	margin: 0;
	padding: 0;
	width: auto;

	&:hover {
		background: none;
	}
`

const Link = styled.a.attrs({
	className: ({ className }) =>
		`ControlButton control-button ${className || ''}`
})`
	align-items: center;
	color: #00aac7;
	cursor: pointer;
	display: inline-flex;
	position: relative;
	text-decoration: none;
`

const Wrapper = ({ href, ...props }) => {
	if (href && href != '') {
		return <Link href={href} {...props} />
	} else {
		return <Button {...props} />
	}
}

const ControlButton = ({
	iconLeft,
	iconRight,
	onClick,
	children,
	...props
}) => (
	<Wrapper
		onClick={onClick}
		iconLeft={iconLeft}
		iconRight={iconRight}
		{...props}
		tabIndex={0}
	>
		{iconLeft && <StyledIconLeft>{iconLeft}</StyledIconLeft>}
		<span>{children}</span>
		{iconRight && <StyledIconRight>{iconRight}</StyledIconRight>}
	</Wrapper>
)

ControlButton.propTypes = {
	onClick: PropTypes.func,
	children: PropTypes.any,
	className: PropTypes.string,
	iconLeft: PropTypes.string,
	iconRight: PropTypes.string
}

export default ControlButton
