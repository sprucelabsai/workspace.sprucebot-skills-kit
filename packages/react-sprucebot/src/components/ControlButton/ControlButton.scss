

.control__button {
  display: inline-flex;
  align-items: center;
  position: relative;
  width: auto;
  margin: 0;
  padding: 0;
  background: none;
  color: #00aac7;
  font-weight: normal;
  letter-spacing: 0;
  text-decoration: none;

  &:hover {
    background: none;
  }

  .icon__left {
    margin-right: 0.25em;
  }

  .icon__right {
    margin-left: 0.25em;
  }
}