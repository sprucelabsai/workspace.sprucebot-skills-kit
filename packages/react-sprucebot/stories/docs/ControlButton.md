# ControlButton

Control Buttons have the option to display a right or left icon
using Material Icons. Check out https://material.io/icons/. To use
an icon, just enter the icon name as a value for the iconLeft or
iconRight prop. You can also pass an href to render the button as a
link!
