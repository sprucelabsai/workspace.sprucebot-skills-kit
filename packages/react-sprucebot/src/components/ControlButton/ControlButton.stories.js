import React from 'react'
import moment from 'moment'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withInfo } from '@storybook/addon-info'
import { withKnobs, boolean, select, text } from '@storybook/addon-knobs/react'
import { withReadme, withDocs } from 'storybook-readme'
import LinkPile from '../LinkPile/LinkPile'
import ControlButton from './ControlButton'
import readme from './ControlButton.md'

const iconOptions = {
	null: 'None',
	favorite: 'Favorite',
	edit: 'Edit',
	close: 'Close',
	keyboard_arrow_up: 'Up',
	keyboard_arrow_down: 'Down',
	chevron_left: 'Left',
	chevron_right: 'Right',
	'ðŸ¤–': 'Bot'
}

const defaultIcon = 'null'

const stories = storiesOf('ControlButton', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(
		readme,
		withInfo()(() => (
			<LinkPile>
				<ControlButton
					onClick={action('ControlButton onClick')}
				>{`Control all the buttons!`}</ControlButton>
				<ControlButton
					iconLeft={text('Left Icon', 'edit')}
					onClick={action('ControlButton onClick')}
				>{`I have a left icon`}</ControlButton>
				<ControlButton
					iconLeft={text('Left Icon', 'edit')}
					iconRight={text('Right Icon', 'favorite')}
					onClick={action('ControlButton onClick')}
				>{`I have a right Icon`}</ControlButton>
				<ControlButton
					iconRight={text('Rigth Icon', 'ðŸ¤–')}
					href={text('Link href', 'https://sprucebot.com')}
					onClick={action('Link onClick')}
				>{`I'm a link!`}</ControlButton>
			</LinkPile>
		))
	)
)
