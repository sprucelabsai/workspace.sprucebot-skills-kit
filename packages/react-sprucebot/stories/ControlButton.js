import React from 'react'
import moment from 'moment'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withInfo } from '@storybook/addon-info'
import { withKnobs, boolean, select, text } from '@storybook/addon-knobs/react'
import { withReadme, withDocs } from 'storybook-readme'
import Container from '../lib/components/Container/Container'
import ControlButton from '../lib/components/ControlButton/ControlButton'
import readme from './docs/ControlButton.md'

const iconOptions = {
	null: 'None',
	favorite: 'Favorite',
	edit: 'Edit',
	close: 'Close',
	keyboard_arrow_up: 'Up',
	keyboard_arrow_down: 'Down',
	chevron_left: 'Left',
	chevron_right: 'Right',
	'ðŸ¤–': 'Bot'
}

const defaultIcon = 'null'

const stories = storiesOf('ControlButton', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(
		readme,
		withInfo()(() => (
			<div className="single_col">
				<ControlButton
					iconLeft={text('Left Icon', 'edit')}
					iconRight={text('Right Icon', 'favorite')}
					onClick={action('Button onClick')}
				>{`Control all the buttons!`}</ControlButton>
			</div>
		))
	)
)
