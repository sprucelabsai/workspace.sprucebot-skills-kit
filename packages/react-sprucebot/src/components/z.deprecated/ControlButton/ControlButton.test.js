import React from 'react'
import ControlButton from './ControlButton'
import { shallow } from 'enzyme'

describe('ControlButton tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<ControlButton />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
