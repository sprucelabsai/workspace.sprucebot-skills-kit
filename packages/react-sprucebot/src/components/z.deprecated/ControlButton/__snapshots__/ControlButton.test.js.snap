// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlButton tests Should match the snapshot 1`] = `
<Button
  alt={false}
  busy={false}
  className="ControlButton control__button "
  hideLoader={true}
  primary={false}
  remove={false}
  secondary={false}
  tabIndex={0}
  tag="button"
  toggle={false}
  type="button"
>
  <span />
</Button>
`;
